### PR TITLE
[LLVM] Add missing pre-processor macro when LLVM is disabled

### PR DESCRIPTION
**Pull request by ghuau-innopeak**
Some components were still referencing LLVM with TI_WITH_LLVM set to
false triggering a build failure.

Related issue = #3679

### CODE REVIEW

**Review by AmesingFlank — APPROVED** [review]
lgtm, thanks!

**Comment by ghuau-innopeak** [comment]
@AmesingFlank Thanks to your feedback, I just pushed the changes related to the other PR I closed to this one as they are all related to LLVM.

File: taichi/backends/device.cpp [hunk 1]
```diff
@@ -29,12 +31,13 @@ Device::MemcpyCapability Device::check_memcpy_capability(DevicePtr dst,
...
 
 #if TI_WITH_VULKAN
+#if TI_WITH_LLVM
   if (dynamic_cast<vulkan::VulkanDevice *>(dst.device) &&
       dynamic_cast<cpu::CpuDevice *>(src.device)) {
     // TODO: support direct copy if dst itself supports host write.
     return Device::MemcpyCapability::RequiresStagingBuffer;
   }
-#if TI_WITH_CUDA
+#elif TI_WITH_CUDA
```
Comment (AmesingFlank) [inline taichi/backends/device.cpp:40]:
```suggestion
#if TI_WITH_CUDA
```

Comment (AmesingFlank) [reply taichi/backends/device.cpp:40]: this is the reason the tests are failing.

Comment (ghuau-innopeak) [reply taichi/backends/device.cpp]: That makes sense, not sure why I used a #elif, I missed this use case in my testing

**Pull request merged**